### PR TITLE
Expose named export (fix TypeScript>=4.8 with native ESM)

### PR DESCRIPTION
**Pull request by greguz**
#### Checklist

- [x] run `npm run test`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -153,6 +155,19 @@ const schema = S.object()
   .valueOf()
 ```
 
+### ESM
+
+A named export is also aviable to work with native ESM modules (aka `Node16` or `NodeNext`):
```
Comment (Fdawgs) [inline README.md:160]:
```suggestion
A named export is also available to work with native ESM modules:
```

**Review by Fdawgs — CHANGES_REQUESTED** [review]

**Comment by Uzlopak** [comment]
@climba03003 
;/

**Review by mcollina — APPROVED** [review]
lgtm

**Review by climba03003 — COMMENTED** [review]
I wouldn't block for a named export with shorten name.

File: src/FluentJSONSchema.js [hunk 2]
```diff
@@ -199,3 +199,5 @@ const fluentSchema = {
 }
 module.exports = fluentSchema
 module.exports.default = fluentSchema
+module.exports.S = fluentSchema
+exports.S = fluentSchema
```
[the remark this answers is not in the document]
Comment (greguz) [reply src/FluentJSONSchema.js:203]: You're right.

**Comment by Inviz** [comment]
Is this a real life? i will finally be able to update TS? LGTM

**Comment by mcollina** [comment]
@Fdawgs @aboutlo PTAL

**Review by Fdawgs — APPROVED** [review]

**Comment by greguz** [comment]
I've noticed a couple of things:
- The `Prettier` badge is wrong, I see this repo uses `Standard` for linting. May I update the badge?
- There's no `files` field inside the `package.json`, so all test files (and much more) are packaged and shipped to `NPM`. May add this field?

Also I've just created a pure-ESM version of this project under the name [fluent-json-schema-es](https://www.npmjs.com/package/fluent-json-schema-es) where you can use the `import S from 'fluent-json-schema-es` syntax freely.

By doing that I saw that a double-module package (bot ESM and CJS code inside the same package) could potentially fix the issue with TypeScript entirely. I suggest [`rollup`](https://rollupjs.org/) to do che ESM>CJS conversion.

What are your thoughts about It? I will be happy to make this change and deprecate `fluent-json-schema-es`.

**Comment by Fdawgs** [comment]
>     * The `Prettier` badge is wrong, I see this repo uses `Standard` for linting. May I update the badge?

Sure, please open a separate PR. 😄 

>     * There's no `files` field inside the `package.json`, so all test files (and much more) are packaged and shipped to `NPM`. May add this field?

See https://github.com/fastify/skeleton/issues/42



> Also I've just created a pure-ESM version of this project under the name [fluent-json-schema-es](https://www.npmjs.com/package/fluent-json-schema-es) where you can use the `import S from 'fluent-json-schema-es` syntax freely.
> 
> By doing that I saw that a double-module package (bot ESM and CJS code inside the same package) could potentially fix the issue with TypeScript entirely. I suggest [`rollup`](https://rollupjs.org/) to do che ESM>CJS conversion.
> 
> What are your thoughts about It? I will be happy to make this change and deprecate `fluent-json-schema-es`.

See https://github.com/fastify/fastify/issues/2847 for ESM discussion.

**Comment by Inviz** [comment]
when's release?

**Pull request merged**